azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1SpecBuilder.java

```diff
@@ -180,6 +182,33 @@ public AzKubernetesV1SpecBuilder addHostPathVolume(final String volName, final S
         return this;
     }
 
+    /**
+     * Adds a volume mount populated from a kubernetes secret.
+     *
+     * @param volName volume name
+     * @param secretName secret name
+     * @param volMountPath directory where secret will be mounted
+     */
+    public AzKubernetesV1SpecBuilder addSecretVolume(final String volName,
+        final String secretName, final String volMountPath) {
+        final V1SecretVolumeSource secretVolumeSource =
+            new V1SecretVolumeSourceBuilder()
+                .withNewSecretName(secretName)
+                .withDefaultMode(0400) //set read permissions only for the the pod user
```

Comment:
Can you please create constant for permission?

Reply:
Fixed.